### PR TITLE
premium; prime the apple uuid from the server

### PR DESCRIPTION
**Pull request by jedireza**
This PR pings the server to receive a UUID associated with the user's account before we call the purchase function on the product.

I'm not 100% sure what problem this solves for the backend, but it might help with some edge cases they've identified.

### How was this tested?
- [x] I tested this manually

### Manual test cases
I tested this with the simulator and observed the calls via the Proxyman app.

### Screenshots and screen recordings
N/A

## Issues addressed
N/A

### CODE REVIEW

File: Client/Frontend/Premium/PremiumStore.swift [hunk 1]
```diff
@@ -59,13 +61,41 @@ class PremiumStore: ObservableObject {
...
             }
-
-            return false
         }
 
         return false
     }
 
+    func primeUUID() async -> String? {
+        await withCheckedContinuation { continuation in
```
Comment (chung-lai) [inline Client/Frontend/Premium/PremiumStore.swift:70]: Not super familiar with this API but looks good and we are calling `resume` in all cases.

**Review by chung-lai — APPROVED** [review]
LGTM
